### ISSUE TITLE
Open3D uses now-removed primary embree `rtcIntersect` API

### ISSUE THREAD

**Issue by pwais** [isl-org/Open3D#6279]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

Open3D uses `rtcIntersect1M`

https://github.com/isl-org/Open3D/blob/6ddbcd5c9b8bf0b496e4151c7d7766af09e3dba7/cpp/open3d/t/geometry/RaycastingScene.cpp#L318C13-L318C27

which has been deprecated and removed from embree:

https://github.com/embree/embree/blob/0fcb306c9176221219dd15e27fe0527ed334948f/README.md?plain=1#L1383

Will perhaps this be addressed when Open3D be moves to the newest embree?

### Steps to reproduce the bug

```python
See links above
```

### Open3D, Python and System information

```markdown
- Operating system: all
- Python version: all
- Open3D version: 0.17.0
- System architecture: all
- Is this a remote workstation?: no
- How did you install Open3D?: pip
- Compiler version (if built from source): all
```

**Comment by benjaminum** [comment]
@pwais Yes, we will update this once we update embree. Unfortunately, we don't have a timeline for that.
